packages/modules/src/dataSelection/index.tsx

```diff
@@ -2,50 +2,37 @@ import React from 'react'
 import {
   ModuleProps,
   useStepInputValue,
-  TabularDataFilter,
   withMockProcessor,
-  TableStats
+  DataRepositoryItemsFilter,
+  useDataRepository,
+  DataRepositoryItemStructure,
+  DataRepositoryItemsTable
 } from '@dharpa-vre/client-core'
-import { List, Table, Utf8 } from 'apache-arrow'
+import { Table } from 'apache-arrow'
 import { TableView } from '../components/TableView'
 
-/**
- * columns in the `corpus` table.
- */
-type CorpusStructure = {
-  // URI of the corpus item.
-  uri: Utf8
-  // Human readable label of the item.
-  label: Utf8
-  // If the corpus item is a tabular file - columns in the file.
-  columns?: List<Utf8>
-}
-type CorpusColumns = keyof CorpusStructure
-
 interface InputValues {
-  repositoryItems?: Table<CorpusStructure>
-  selectedItemsUris?: string[]
-  metadataFields?: CorpusColumns[]
+  repositoryItems?: Table<DataRepositoryItemStructure>
+  selectedItemsIds?: string[]
+  metadataFields?: string[]
 }
 
 interface OutputValues {
-  selectedItems?: Table<Partial<CorpusStructure>>
+  selectedItems?: Table<Partial<DataRepositoryItemStructure>>
 }
 
 type Props = ModuleProps<InputValues, OutputValues>
 
 const DataSelection = ({ step }: Props): JSX.Element => {
-  const [repositoryItemsFilter, setRepositoryItemsFilter] = React.useState<TabularDataFilter>({ pageSize: 5 })
-  const [selectedItemsUris = [], setSelectedItemsUris] = useStepInputValue<string[]>(
+  const [repositoryItemsFilter, setRepositoryItemsFilter] = React.useState<DataRepositoryItemsFilter>({
+    pageSize: 5
+  })
+  const [selectedItemsIds = [], setSelectedItemsIds] = useStepInputValue<string[]>(
     step.stepId,
-    'selectedItemsUris'
+    'selectedItemsIds'
   )
   const [metadataFields = [], setMetadataFields] = useStepInputValue<string[]>(step.stepId, 'metadataFields')
-  const [repositoryItemsBatch, , tableStats] = useStepInputValue<Table<CorpusStructure>, TableStats>(
-    step.stepId,
-    'repositoryItems',
-    repositoryItemsFilter
-  )
+  const [repositoryItemsBatch, repositoryStats] = useDataRepository(repositoryItemsFilter)
```

Comment:
That's how pages from data repository can be retrieved. The filter contains only offset and page size fields, but we will add more when needed. E.g. filtering by alias, data type, etc..